### PR TITLE
fix: add multiple email check

### PR DESCRIPTION
**Pull request by Merloss**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: api/src/routes/auth.ts [hunk 1]
```diff
@@ -32,14 +32,16 @@ const getEmailFromAuth0 = async (req: FastifyRequest) => {
...
-  // TODO: handle the case where there are multiple users with the same email.
-  // e.g. use findMany and throw an error if more than one is found.
-  const existingUser = await fastify.prisma.user.findFirst({
+  const existingUsers = await fastify.prisma.user.findMany({
     where: { email },
     select: { id: true }
   });
+
+  if (existingUsers.length > 1) 
+    throw new Error("Multiple Users Found");
```
Comment (naomi-lgbt) [inline api/src/routes/auth.ts:41]:
```suggestion
  if (existingUsers.length > 1) throw new Error('Multiple Users Found');
```

Looks like prettier expects this to be on the same line.

**Comment by raisedadead** [comment]
Hi there, 

Thanks for creating this pull request.

Please ensure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously. Respect the requirements to earn the respect of your fellow maintainers. You will also save everyone time :)

Since you are new here, please pace yourself and read our [contributing guidelines](https://contribute.freecodecamp.org), as we see that you may still need to catch up on a few things.

<details>
<summary>Here are some examples (expand)</summary>
 
1. Do not edit files directly on GitHub – while you can, it's not a good idea. 

   Typos and formatting errors can break the tests.

2. Use the correct way to link issues. 

   Add the issue number only in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.

4. Keep mentions and review requests to a minimum. 

   We understand you are excited about contributing, and our maintainers will get back to you when they can.

5. Do not work directly off your `main` branch. 

   You can always create a new branch for the changes you are working on. That way, you can sync change to your PR branch as the main repository moves ahead while your PR is waiting in the merge queue.

</details>

Don't worry. You don't have to close this PR. Feel free to ask specific queries on improving your PR here; someone will guide you.

We are happy you are excited to contribute and appreciate you taking the time to help us. Looking forward to more contributions!

Happy Contributing.

**Comment by Merloss** [comment]
@raisedadead @naomi-lgbt any updates? Is there a problem?

**Comment by raisedadead** [comment]
> any updates? Is there a problem?

Yes - unfortunately, it isn't easy to review the code as is and may need more involved approach because:

- It was edited directly off of GitHub and presumably was never tested locally.
- The PR is not expanding or updating any tests, so we don't know the effects.
- Finally, We need clarification on what the code is doing. 

  The PR description needs to explain more. It's more like, "Here is the title and some code, and please make sense of it on your own". That is not ideal for maintainers.

All this makes reviewing difficult because we need more context around the suggested changes. You can help by addressing the above – someone will eventually review this when they have more time. You are welcome to join us in the chat to ask more questions if you like.

Happy Contributing!

**Comment by Merloss** [comment]
```
// TODO: handle the case where there are multiple users with the same email.
// e.g. use findMany and throw an error if more than one is found.
```
I just completed the [TODO](https://github.com/freeCodeCamp/freeCodeCamp/pull/51190/commits/28a974102c4afbeec44a08268ccad924463990ef#diff-d051e648616a56c5ea6675b37ba577d918cf4a2a618448d115bd0519cac5e7eeL35) listed above. No need to explain, in my opinion.

**Comment by ShaunSHamilton** [comment]
Hello there,

Thank you for taking the time on this. However, I am not sure we are going to take this change just yet.

We are porting over our current `api-server` to a new `api`, and, for the most part, are making a one-to-one copy for the MVP. When we notice bugs from the old API, we add TODOs for a later version.

Hope that makes sense.

If someone else could confirm. _I will get fully back to all the API stuff, in the next week or so_

**Pull request closed without merging**